### PR TITLE
Initial Commit for Community Readme Docs

### PR DESCRIPTION
**Pull request by v1k0d3n**
Starting to get a TOC formed that captures some of the input provided in [etherpad](https://etherpad.openstack.org/p/openstack-helm-standards). This makes it available for others to commit readme documentation to  the repo, with some relative structure.

---
This change is [Reviewable](https://reviewable.io/reviews/att-comdev/openstack-helm/146)

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+## Table of Contents
+
+1. [Openstack-Helm Design Principals]()
+  1.1 [Mission]()
+    1.1.1 [Resiliency]()
+    1.1.2 [Scaling]()
+  1.2 [Helm Overrides]()
+  1.3 [Init-Containers]()
+  1.4 [Kubernetes Jobs]()
+  1.5 [Dependency Checking]()
```
Comment (alanmeadows) [inline docs/README.md]: Aren't these subsections of 1.3? We seem wedded to init containers to help with this.

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  1.6 [Service Registration]()
```
Comment (alanmeadows) [inline docs/README.md]: Same as above.  This is part of init-containers.

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  1.7 [Resource Limits]()
+  1.8 [Conditionals]()
```
Comment (alanmeadows) [inline docs/README.md]: This seems generic, whats an example of a conditional? Development mode? I think Development Mode (and its intentions) deserves its own section.

File: docs/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  4.1 [Network Policies]()
+  4.2 [Advanced Network Policies]()
+  4.3 [Role-Based Access Controls]()
+  4.4 [Security Contexts]()
+  4.5 [Security Add-Ons]()
+5. [Developer Resources](https://github.com/att-comdev/openstack-helm/tree/master/docs/developer)
+  5.1 [Contributions and Guidelines]()
+  5.2 [Development Tools]()
+    5.2.1 [Minikube Development](https://github.com/att-comdev/openstack-helm/blob/master/docs/developer/minikube.md)
+  5.3 [Tips and Considerations]()
```
Comment (alanmeadows) [inline docs/README.md]: What about ingress philosophy?

Comment (alanmeadows) [inline docs/README.md]: ... or a networking section which could house that.

File: docs/README.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+1. [Openstack-Helm Design Principals]()
+
+  1.1 [Mission]()
+
+    1.1.1 [Resiliency]()
+
+    1.1.2 [Scaling]()
+
+  1.2 [Helm Overrides]()
+
```
Comment (alanmeadows) [inline docs/README.md]: I'd like to see a "Single File Site Definition Overrides"

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    1.2.1 [Resource Limits]()
+
+    1.2.2 [Conditionals]()
+
+  1.3 [Init-Containers]()
```
Comment (alanmeadows) [inline docs/README.md]: Parallel to Helm Overrides: Helm Upgrades

File: docs/README.md (hunk 5 above)
Comment (v1k0d3n) [inline docs/README.md]: "Single Chart" I think is better....

File: docs/README.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+    1.4.1 [Service Registration]()
+
+    1.4.2 [User Registration]()
+
+    1.4.3 [Database Creation]()
+
+    1.4.4 [Database Migration]()
+
+  1.5 [Complimentary Efforts]()
```
Comment (alanmeadows) [inline docs/README.md]: We need a section to cover what we just discussed: Kubernetes State (thirdpartyresources for anything we develop internally, like galera management), and we sanction things like upstream rabbitmq using etcd for clustering/state if upstream says thats their best practice.

**Pull request merged**